api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupPNCRegister.java

```diff
@@ -0,0 +1,243 @@
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * Created by Nicholas Ingosi on 5/10/17.
+ */
+@Component
+public class SetupPNCRegister extends UgandaEMRDataExportManager {
+
+    @Autowired
+    private DataFactory df;
+
+    @Autowired
+    SharedDataDefintion sdd;
+
+    /**
+     * @return the uuid for the report design for exporting to Excel
+     */
+    @Override
+    public String getExcelDesignUuid() {
+        return "6625672c-3ee1-11e7-ad03-507b9dc4c741";
+    }
+
+    @Override
+    public List<ReportDesign> constructReportDesigns(ReportDefinition reportDefinition) {
+        List<ReportDesign> l = new ArrayList<ReportDesign>();
+        l.add(buildReportDesign(reportDefinition));
+        return l;
+    }
+
+    /**
+     * Build the report design for the specified report, this allows a user to override the report design by adding properties and other metadata to the report design
+     *
+     * @param reportDefinition
+     * @return The report design
+     */
+    @Override
+    public ReportDesign buildReportDesign(ReportDefinition reportDefinition) {
+        ReportDesign rd = createExcelTemplateDesign(getExcelDesignUuid(), reportDefinition, "PNCRegister.xls");
+        Properties props = new Properties();
+        props.put("repeatingSections", "sheet:1,row:8-10,dataset:PNC");
+        props.put("sortWeight", "5000");
+        rd.setProperties(props);
+        return rd;
+    }
+
+    @Override
+    public String getUuid() {
+        return "489baf0e-3ee1-11e7-a1d1-507b9dc4c741";
+    }
+
+    @Override
+    public String getName() {
+        return "Integrated Postnatal Register";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Contains mothers information after delivery";
+    }
+
+    @Override
+    public ReportDefinition constructReportDefinition() {
+        ReportDefinition rd = new ReportDefinition();
+
+        rd.setUuid(getUuid());
+        rd.setName(getName());
+        rd.setDescription(getDescription());
+        rd.addParameters(getParameters());
+        rd.addDataSetDefinition("PNC", Mapped.mapStraightThrough(dataSetDefinition()));
+        return rd;
+    }
+
+    @Override
+    public String getVersion() {
+        return "0.1";
+    }
+
+    @Override
+    public List<Parameter> getParameters() {
+        List<Parameter> l = new ArrayList<Parameter>();
+        l.add(df.getStartDateParameter());
+        l.add(df.getEndDateParameter());
+        return l;
+    }
+
+    private DataSetDefinition dataSetDefinition() {
+        PatientDataSetDefinition dsd = new PatientDataSetDefinition();
+        dsd.setName("PNC");
+        dsd.addParameters(getParameters());
+        dsd.addRowFilter(Cohorts.genderAndHasAncEncounter(true, false, "fa6f3ff5-b784-43fb-ab35-a08ab7dbf074"), "startDate=${startDate},endDate=${endDate}");
+
+        PersonAttributeType attribute = Context.getPersonService().getPersonAttributeTypeByUuid("14d4f066-15f5-102d-96e4-000c29c2a5d7");
```

Comment:
Rename the attribute to the name of the attribute being loaded to ease identification